cypress/integration/validation/profile.js

```diff
@@ -0,0 +1,241 @@
+const { debug } = require("console")
+
+describe('Checking Profile', () => {
+    beforeEach(() =>{
+        cy.loginViaApi("max@test.test", "password")
+
+        cy.intercept("GET",/.*settings.*/).as('get')
+        cy.visit("/my-profile")
+        cy.wait("@get")
+
+        cy.get('@get')
+        .its("response")
+        .should("include",{
+            statusCode: 200
+        })
+
+
+    })
+    it('trys to validate data being displayed on profile', () =>{
+
+        cy.get("input[id='username']")
+        .should('have.value', 'test_user')
+        cy.get("input[id='email']")
+        .should('have.value', "max@test.test")
+    })
+    it('should NOT be able to change profilename (wrong password)', () =>{
+
+        cy.contains("Bearbeiten")
+        .click()
+
+        //username
+        cy.get("input[id='username']")
+        .clear()
+        .type("test_user1")
+
+        cy.get("input[id='current_password']")
+        .clear()
+        .type("passwort") //invalid
+
+        cy.intercept("POST", /.*api\/users.*/).as("user")
+        cy.contains("Änderungen speichern")
+        .click()
+        cy.wait("@user")
+        .its("response")
+        .should('include',{
+            statusCode: 401
+          })
+
+          
+        cy.intercept("GET",/.*settings.*/).as('get')
+        cy.reload();
```

Comment:
check if neccessary

Reply:
whats your result of checking if this is neccessary? Please comment your results

Reply:
It's for the purpse to get the former values back in that moment.

` //username
        cy.get("input[id='username']")
        .clear()
        .type("test_user1")

        cy.get("input[id='current_password']")
        .clear()
        .type("passwort") //invalid

        cy.intercept("POST", /.*api\/users.*/).as("user")
        cy.contains("Änderungen speichern")
        .click()
        cy.wait("@user")
        .its("response")
        .should('include',{
            statusCode: 401
          })

          
        cy.intercept("GET",/.*settings.*/).as('get')
        cy.reload();
        cy.wait("@get")
        cy.get('@get')
        .its("response")
        .should("include",{
            statusCode: 200
        })
    cy.get("input[id='username']")
        .should('have.value',"test_user")
`

In this case cypress try's to change the username, it should fail, i reload the page and check if the username has changed.
It also reset's the form to its true values.